src/zodbbrowser/references.py

```diff
@@ -0,0 +1,136 @@
+
+import sqlite3
+
+from ZODB.utils import u64
+from ZODB.serialize import referencesf
+from zope.interface import implements
+
+from zodbbrowser.interfaces import IReferencesDatabase
+
+
+def connect(callback):
+    """Decorator for the reference database to access the sqlite DB.
+    """
+
+    def wrapper(self, *args, **kwargs):
+        try:
+            connection = sqlite3.connect(self.db_name)
+        except:
+            raise ValueError('impossible to open references database')
+        try:
+            result = callback(self, connection, *args, **kwargs)
+        except:
+            connection.close()
+            raise
+        return result
+
+    return wrapper
+
+
+class ReferencesDatabase(object):
+    implements(IReferencesDatabase)
+
+    def __init__(self, db_name=':memory:'):
+        self.db_name = db_name
+
+    @connect
+    def analyzeRecords(self, connection, records):
+        cursor = connection.cursor()
+        for record in records:
+            current_oid = u64(record.oid)
+            referred_oids = set(map(u64, referencesf(record.data)))
+
+            for referred_oid in referred_oids or [-1]:
+                cursor.execute("""
+INSERT INTO links (source_oid, target_oid) VALUES
+({0}, {1})
+            """.format(current_oid, referred_oid))
```

Comment:
I don't like sudden indentation drops in the middle of a method.

SQL ignores extra whitespace, so you could align these.  Or you could use `textwrap.dedent()` if you want to see pretty SQL statements in debug log files (assuming sqlite lets you do that).

Also I'd strongly prefer prepared statements instead of `str.format()`, even if it *probably* won't lead to SQL injection here.